File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.py
def solution(k, score):
    answer = []
    result = []
    for i in range(len(score)):
        if len(answer) < k:
            answer.append(score[i])
            result.append(min(answer))
        else:
            if min(answer) < score[i]:
                answer.append(score[i])
                answer.remove(min(answer))
                result.append(min(answer))
            else:
                result.append(min(answer))
    
    return result

File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.java
import java.util.Collections;
import java.util.ArrayList;

class Solution {
    public int[] solution(int k, int[] score) {
        
        ArrayList<Integer> list = new ArrayList<>();
        int[] arr = new int[score.length];
        
        for(int i=0; i<score.length; i++) {
            if (list.size() < k) {
                list.add(score[i]);
            } else {
                if(Collections.min(list) < score[i]) {
                    list.remove(Collections.min(list));
                    list.add(score[i]);
                }
            }
            arr[i] = Collections.min(list);
        }
        
        return arr;
    }
}